utils/cd_version_strings.sh

```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+# Adjusts the version strings in main.cpp, credits.txt and EndlessSky-Info.plist for a nightly build.
+
+perl -p -i -e 's/(Endless Sky) [\d.]+/$1 ${{ github.sha }}/ig' source/main.cpp
```

Comment:
Can we make this script just "set-version" rather than "cd-version-strings" and take an argument, rather than requiring it to be run only from the actions environment (where `${{ }}` does something)?

Then if called with the current version, running `git diff` would result in no changes.

(The line which updates the credits string would probably not be something included, as that's very specific to the CD workflows. A script which just sets versions, and a script which updates credits strings and calls the set version would be viable, too.)

My reasoning here is that - since we will export utils, they should be generally useful too

Reply:
The simplest solution imo would be to just call git instead of the convenient action variables:
```bash
perl -p -i -e "s/(Endless Sky) [\d.]+/\$1 $(git rev-parse --verify HEAD)/ig" source/main.cpp
```

> My reasoning here is that - since we will export utils, they should be generally useful too

I don't disagree, but we shouldn't be making up usecases. I can't think of a case where this script is used other than some kind of bleeding-edge build. The main reason that this is a utils script is because it's used 4 times in the actions job.

Reply:
Don't consider it "making up" usecases, more "enabling". Any fork or contributor or collaborator (or non-GitHub CI) that pulls in this script would then be able to do something as simple as `./utils/set-version "my cool version"` (provided of course that `perl` is available in the executing context).

Reply:
I see what you're going for. I'll also remove the change to the .plist file, since i just noticed no sha256 can conform to apple's [`CFBundleVersion` spec](https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html#//apple_ref/doc/uid/20001431-102364).

Reply:
Can we just append "a" or "d" to the CFBundleShortVersion and set that as the CFBundleVersion, as a part of the cd's script?

There's one other change that i think the set-version script should do, and that's update the man file with the date and version:
```
.TH endless\-sky 6 "24 Feb 2020" "ver. 0.9.11" "Endless Sky"
```

Reply:
The man file shouldn't be a problem. 

About XCode, some research: [`CFBundleShortVersionString`](https://developer.apple.com/documentation/bundleresources/information_property_list/cfbundleshortversionstring) is the actual version, e.g `0.9.11`, and needs to be in `x.y.z` format, no exceptions (not even letters). [`CFBundleVersion`](https://developer.apple.com/documentation/bundleresources/information_property_list/cfbundleversion) also only allows integers, but is intended for build numbers, so `1234` is valid.

We can abuse that by not modifying `CDBundleShortVersionString`, but incrementing `CDBundleVersion` (e.g. by setting it to the actions run ID, or the amount of commits).